lib/handler/redirect-handler.js

```diff
@@ -227,9 +227,10 @@ function cleanRequestHeaders (headers, removeContent, unknownOrigin) {
       }
     }
   } else if (headers && typeof headers === 'object') {
-    for (const key of Object.keys(headers)) {
+    const entries = headers instanceof Headers ? headers.entries() : Object.entries(headers)
```

Comment:
```suggestion
    const entries = Object.getPrototypeOf(headers) === Headers ? headers.entries() : Object.entries(headers)
```

Reply:
Don't exactly understand why but that option does not work, so I stuck with the instanceof let me know if it's good to you.

Reply:
Hmm, I think i know why but let's keep it as is then 👍